docs/products/clickhouse/concepts/service-architecture.rst

```diff
@@ -0,0 +1,99 @@
+
+.. _astacus:
+
+Astacus
+'''''''
+
+Astacus is an open-source project originated at Aiven for coordinating backups of cluster databases, including ClickHouse.
+
+Data architecture
+-----------------
+
+Aiven for ClickHouse enforces
+
+* Full schema replication: all databases, tables, users, and grants are the same on all nodes.
+* Full data replication: table rows are the same on all nodes within a shard.
+
+:ref:`Astacus <astacus>` does most of the backup and restore operations.
+
+Engines: database and table
+---------------------------
+
+ClickHouse has engines in two flavors: table engines and database engines.
+
+* Database engine is responsible for manipulating tables and decides what happens when you try to list, create, or delete a table. It can also be used to restrict a database to specific table engine or to manage replication.
+* Table engine decides how to store data on a disk or how to enable reading data from outside the disk to expose it as a virtual table.
+
+.. _replicated-database-engine:
+
+``Replicated`` database engine
+''''''''''''''''''''''''''''''
+
+The default ClickHouse database engine is the Atomic engine, responsible for creating table metadata on the disk and configuring which table engines are allowed in each database.
+
+Aiven for ClickHouse uses the ``Replicated`` database engine, which is a variant of Atomic. With this engine variant, queries for creating, updating, or altering tables are replicated to all other servers using :ref:`ZooKeeper <zookeeper>`.
+As a result, all servers can have the same table schema, which makes them an actual data cluster and not multiple independent servers that can talk to each other.
+
+.. _replicated-table-engine:
+
+``Replicated`` table engine
+'''''''''''''''''''''''''''
+
+The table engine is responsible for the INSERT and SELECT queries. From a wide variety of available table engines, the most common ones belong to the ``MergeTree`` engines family, which is supported in Aiven for ClickHouse.
+
+.. seealso::
+
+    For a list of all the table engines that you can use in Aiven for ClickHouse, see :doc:`Supported table engines in Aiven for ClickHouse </docs/products/clickhouse/reference/supported-table-engines>`.
+
+``MergeTree`` engine
+~~~~~~~~~~~~~~~~~~~~
+
+With the ``MergeTree`` engine, at least one new file is created for each INSERT query and each new file is written once and never modified. In the background, new files (called *parts*) are re-read, merged, and rewritten into compact form. Writing data in parts determines the performance profile of ClickHouse.
+
+* INSERT queries need to be batched to avoid handling a number of small parts.
+* UPDATE and DELETE queries need to be batched. Removing or updating a single row requires rewriting an entire part with all the rows except the one we want to remove or update.
+* SELECT queries are executed fast because all the data found in a part is valid and all files can be cached since they never change.
```

Comment:
```suggestion
* SELECT queries are executed rapidly because all the data found in a part is valid and all files can be cached since they never change.
```